skpro/classification/sklearn_classifiers/_sklearn_proba_class.py

```diff
@@ -0,0 +1,144 @@
+"""Adapter to sklearn multiclass classification using Histograms."""
+
+__author__ = ["ShreeshaM07"]
+
+import numpy as np
+import pandas as pd
+
+from skpro.distributions import Histogram
+from skpro.regression.base import BaseProbaRegressor
+from skpro.utils.sklearn import prep_skl_df
+
+
+class SklearnProbaClassifier(BaseProbaRegressor):
+    """A multiclass classifier fitting a histogram distribution."""
+
+    _tags = {
+        "authors": ["ShreeshaM07"],
+        "maintainers": ["ShreeshaM07"],
+        "capability:multioutput": False,
+        "capability:missing": True,
+    }
+
+    def __init__(self, clf, bins=10):
+        self.clf = clf
+        if isinstance(bins, int) or isinstance(bins, np.integer):
+            bins = np.arange(bins + 1)
+        self.bins = bins
+
+        super().__init__()
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from sklearn import clone
+
+        self.clf_ = clone(self.clf)
+        self._y_cols = y.columns
+
+        X = prep_skl_df(X)
+        y = prep_skl_df(y)
+
+        if isinstance(y, pd.DataFrame) and len(y.columns) == 1:
```

Comment:
some binning should happen here

Reply:
I'll be looking into that next. But I am not quite sure what we want to bin here. Could you please elaborate a little on your thought.

Reply:
The estimator should implement the algorithm from the issue description: https://github.com/sktime/skpro/issues/378#issue-2341097717

In `_fit`, that means:

1. bin the regression target `y` into bins, according to `bins`. This results in a multioutput target `y_binned`.
2. Fit the classifier to `X` and `y_binned`.

Reply:
> What I would infer from this is
> if y is a dataframe with the output classes

I think there is a bit of a miscommunication, sorry for not explaining clearly.

The `y` arriving in `_fit` will contain floats, not categories, for instance

```
       y
0  5
1  2
2  6
3  3
```

Let's say we decide to bin with `bins = [1,4, 7]`, then internally we convert this to


```
       y_binned
0  "class2"
1  "class1"
2  "class2"
3  "class1"
```

to this we fit `clf`.

In `_predict_proba`, using `clf.predict_proba` will produce sth like


```
      class1    class2
0   0.6          0.4
1   0.3          0.7
2   0.1          0.9
```

And this we then convert to an 3 times 1 `Histogram` distribution.

Reply:
Ah I get it now thanks for the clarification.

Reply:
But again with the `int` case for `bins` we will be unclear as to where to start and end the bins array. So would it be a good idea to take it as a tuple again (bins start,bins end,number of bins) like in the Histogram distribution? Else if we want to keep it as `int` itself then I would have to do `bins` = `(min(X),max(X),number of bins inputted by user)`.

Reply:
I think having the `int` option with min/max would enhance user experience, especially if it also has a default.

Re min/max, why not directly min/max, but 0.999 times etc?

Reply:
That is to include the min(X) and max(X) values in the bins and not to have them as the bin boundary. It is done in the same way as `pandas.cut` in case of `int`.

`int` : Defines the number of equal-width bins in the range of x. The range of x is extended by .1% on each side to include the minimum and maximum values of x.

Reply:
I have rethought about it and since I will be using `np.arange` to create the bins array in the `int` case it would be best to take it as `np.arange(min(y),maxy(y)+step,step)` where `step = (max(y)-min(y))/bins`.

Reply:
I think this is not a good idea, using `np.arange` where `stop - start` is an approximate multiple of `step` leads to a length of the output which is not numerically stable, see the docs of `np.arange` which warn exactly of that.

Instead, I would simply use `np.linspace` with `num=bins` (in the integer case).

Another useful function to be used later on is `np.digitize`.

Reply:
Yes I have used `np.linspace` now it makes more sense and is more accurate. I will still have to do start = 0.999*min(y)and stop = max(y)*1.001 as the `pandas.cut`, which I am using to bin the y values, does not include the leftmost point of the bin so I have to increase the domain in order to consider the min(y) value.